src/query.ts

```diff
@@ -138,6 +138,31 @@ export class Query<T> {
     );
   }
 
+  /**
+   * Add sql statements to the batch only
+   *
+   * @returns The parameters for the batch
+   */
+  public addToBatch(parameters: BindingValue[]): BindingValue[] {
+    this.parameters = this.parameters ?? [];
+    if (!Array.isArray(parameters)){

```

Comment:
In theory we don't need to type check. Language server will catch the type errors, no matter if JavaScript or TypeScript is used.

Reply:
I will keep it here to provide a clearer error message to the user.

Reply:
Sounds good. Perhaps the logic can be improved:

```ts
const isError = Array.isArray(parameters) : false ? parameters.some(p => !Array.isArray(parameter))
if (isError) throw new Error(`Parameter 'parameters' must be a 2d array of parameters to the query`);
```